programs/mango-v4/src/health/account_retriever.rs

```diff
@@ -227,12 +259,13 @@ impl<'a, 'info> ScannedBanksAndOracles<'a, 'info> {
     ) -> Result<(&mut Bank, I80F48, Option<(&mut Bank, I80F48)>)> {
         if token_index1 == token_index2 {
             let index = self.bank_index(token_index1)?;
+            let price = {
+                let bank = self.banks[index].load_fully_unchecked::<Bank>()?;
+                let oracle_acc_infos = self.create_oracle_infos(index, Some(&bank.fallback_oracle));
+                bank.oracle_price_with_fallback(&oracle_acc_infos, self.staleness_slot)?
+            };
```

Comment:
Since these function take in `&mut self` (where self is some sort of AccountRetriever) `let bank = self.banks[index].load_mut_fully_unchecked::<Bank>()?;` is a mutable borrow. At the same time`self.create_oracle_infos(index, Some(&bank.fallback_oracle));` is immutable.

I decided to drop the immutable borrow before the mutable one to make the borrow checker happy but I'm open to alternatives. This was the least-invasive way I could think of

Reply:
Ah yes, unfortunate but necessary. Agree this is the simplest way to do it when faced with mutable borrows like this.